### PR TITLE
fix: remove targets not found in config

### LINKED ISSUE #322: Remove all targets before adding targets for a new ceremony
**Description**

**Version**

Old targets like rekor.0.pub were left around.

### PR DESCRIPTION
**Pull request by cdris**
Signed-off-by: Ceridwen Driskill <cdriskill@google.com>

#### Summary

Fixes #322, removes targets not specified in the given config.

### CODE REVIEW

**Comment by cdris** [comment]
@asraa and @haydentherapper since I can't add reviewers myself

File: cmd/tuf/app/init.go [hunk 1]
```diff
@@ -205,6 +206,24 @@ func InitCmd(ctx context.Context, directory, previous string, threshold int, tar
...
+	if err != nil {
+		return err
+	}
+	for path := range allTargets {
+		if !expectedTargets[path] {
+			targetsToRemove = append(targetsToRemove, path)
+		}
+	}
+	if len(targetsToRemove) > 0 {
+		if err := repo.RemoveTargetsWithExpires(targetsToRemove, getExpiration("targets")); err != nil {
```
Comment (asraa) [inline cmd/tuf/app/init.go:226]: interesting! does `RemoveTargetsWithExpires` with an empty slice of targets end up removing all targets (is this what you were mentioning?)

if so can you add a quick comment like "only remove with non-empty targets. passing an empty slice removes all targets"

File: tests/e2e_test.go [hunk 2]
```diff
@@ -584,3 +584,184 @@ func TestRotateRootKey(t *testing.T) {
...
+	// Verify with go-tuf
+	targetFiles, err = verifyGoTuf(t, td, meta["root.json"])
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(targetFiles) != 1 {
+		t.Fatalf("expected one target, got %d", len(targetFiles))
+	}
+	for name := range targetFiles {
+		if !strings.EqualFold(name, "bar.txt") {
```
Comment (asraa) [inline tests/e2e_test.go:763]: nice!!!

**Review by asraa — DISMISSED** [review]
Thank you so much! And thank you for adding the test! LGTM

File: cmd/tuf/app/init.go (hunk 1 above)
Comment (cdris) [reply cmd/tuf/app/init.go:226]: That's exactly what it does, will add a comment.

**Review by Hayden-IO — DISMISSED** [review]
Nice work!

**Review by Hayden-IO — APPROVED** [review]

**Pull request merged**